httpbp/headers.go

```diff
@@ -58,14 +58,16 @@ const (
 
 type headerContextKey HeaderContextKey
 
-func setHeader(ctx context.Context, key HeaderContextKey, value string) context.Context {
+// SetHeader sets the value on the context at key.
+func SetHeader(ctx context.Context, key HeaderContextKey, value string) context.Context {
```

Comment:
I don't think we want to export this function, this opens a can of worms we don't want to deal with.

I think the only reason you want to export this is to be used in the test of edgecontext? Can we move that test into this package instead? Or by constructing the actual http headers?

Reply:
On second thought, this is OK. We already do that in thrift library.

But in that case we no longer need the extra unexported type of `headerContextKey`, we should just use `HeaderContextKey` as the context key directly.